packages/server/src/api/controllers/application.ts

```diff
@@ -293,11 +304,25 @@ async function performAppCreate(
   return await context.doInAppContext(appId, async () => {
     const instance = await createInstance(appId, instanceConfig)
     const db = context.getAppDB()
+    const isImport = !!instanceConfig.file
+    const addSampleData = !isImport && !useTemplate
 
     if (instanceConfig.useTemplate && !instanceConfig.file) {
       await updateUserColumns(appId, db, ctx.user._id!)
     }
 
+    // Add link to sample data page if required
+    const newLinks = addSampleData
```

Comment:
Could we perhaps break out the navigation updates into an `addSampleNavigation` function like we have for the others parts of the sample data? I was looking for a minute trying to work out where the screen got added to the nav.

Reply:
The links are part of the app metadata, so if I break them out into a function then I'd need to do another update of the metadata after the app is created or something which is wasteful, since I can just add it in the initial doc that is created.

Either that or just have a function that mutates the metadata passed in and adds that one link, but that seems a little convoluted... but would still move it out at least. I'll add that if you'd prefer it. This at least won't do another couch read/write.

Reply:
I don't necessarily think that doing a separate write is a bad idea, my reasoning being that the navigation has been updated here before the data/screens has been added yet, in the same vein as my previous comment, the app is going to exist even if the sample data/screens fail to be created, so theres a potential scenario right now where the navigation has a link to a screen that does not exist yet.

Doing this as a separate update after the data/screens has been added in the `if (addSampleData) ...` block means that if something fails there, the nav will be the last thing to be updated, so worst case the nav will never have links to non-existent screens added to it.